### PR TITLE
Avoid `nodeGroups[0].name must be set` validation error

### LINKED ISSUE #1164: get nodegroups command no longer display all nodegroups
**What happened?**
You can no longer get all nodegroups via a single command. You can only get individual nodegroup settings. 

**What you expected to happen?**
Expect to display all nodegroups without entering a nodegroup name, which was capable in v0.3.1. At least another flag/option to display all groups. 

**How to reproduce it?**
eksctl get nodegroup --cluster amg-non-prod-use1
[✖]  nodeGroups[0].name must be set

**Versions**
```
eksctl version
[ℹ]  version.Info{BuiltAt:"", GitCommit:"", GitTag:"0.4.0"}
```

### PR DESCRIPTION
**Pull request by errordeveloper**
### Description

This fixes #1164 and related cases of the same error. This is a regression that was introduced by  #1116 and #1124.

### Checklist

- [x] Code compiles correctly (i.e `make build`)
- [x] Added tests that cover your change (if possible)
- [x] All unit tests passing (i.e. `make test`)
- [x] Manually tested

### CODE REVIEW

**Comment by errordeveloper** [comment]
I'm, going to have a go at adding some unit test to cover this, possibly also integration tests.

File: pkg/ctl/get/nodegroup.go [hunk 1]
```diff
@@ -62,6 +54,20 @@ func doGetNodeGroup(cmd *cmdutils.Cmd, ng *api.NodeGroup, params *getCmdParams)
 		ng.Name = cmd.NameArg
 	}
 
+	if ng.Name == "" {
+		// avoid validation errors when listing all nodegroups
+		cfg.NodeGroups = nil
```
Comment (cPu1) [inline pkg/ctl/get/nodegroup.go]: While this skips the validation for node groups, it'd be ideal if we could avoid calling `ValidateClusterConfig` for this usage of the command as there's no node group input to validate here.

Comment (errordeveloper) [reply pkg/ctl/get/nodegroup.go]: At the moment the idea is that `ValidateClusterConfig` is mean to be call in all instances, whether `ClusterConfig` instance was constructed by the command, or parsed from a config file. I would rather review how it's constructed. I'll see if there is an easy fix to this that is a more coherent.

**Comment by errordeveloper** [comment]
@cPu1 what do you make of most recent commit?

**Comment by cPu1** [comment]
>@cPu1 what do you make of the most recent commit?

LGTM

The unit test is failing because the change to `NewCreateClusterLoader` isn't updated in `configfile_test.go:126`

**Comment by errordeveloper** [comment]
Yes, I'll fix the unit tests and try to add some new ones also....

**Comment by errordeveloper** [comment]
I'd like to add `eksctl get ng` to integration suite as well.

File: pkg/ctl/cmdutils/configfile_test.go [hunk 2]
```diff
@@ -123,11 +123,95 @@ var _ = Describe("cmdutils configfile", func() {
...
+					Expect(cmd.ClusterConfig.NodeGroups[0]).To(Equal(loaderTest.ng))
+				}
+
+				_, err := cmd.NewCtl()
+				Expect(err).ToNot(HaveOccurred())
+			}
+		})
+
+		It("loader should handle nodegroup exclusion with config file", func() {
+			loaderParams := []struct {
```
Comment (cPu1) [inline pkg/ctl/cmdutils/configfile_test.go]: Alternative way:

```golang
		loaderParams := []struct {
			configFile     string
			nodeGroupCount int
		}{
			{"01-simple-cluster.yaml", 1, true},
			{"02-custom-vpc-cidr-no-nodes.yaml", 0, true},
			{"03-two-nodegroups.yaml", 2, true},
			{"05-advanced-nodegroups.yaml", 3, true},
			{"07-ssh-keys.yaml", 5, true},
		}
		
		for _, loaderTest := range loaderParams {
			cmd := &Cmd{
				CobraCommand:      newCmd(),
				ClusterConfigFile: filepath.Join(examplesDir, loaderTest.configFile),
				ClusterConfig:     api.NewClusterConfig(),
				ProviderConfig:    &api.ProviderConfig{},
			}
		
			ngFilter := NewNodeGroupFilter()
		
			for _, withoutNodeGroup := range []bool{true, false} {
				Expect(NewCreateClusterLoader(cmd, ngFilter, nil, withoutNodeGroup).Load()).To(Succeed())
		
				Expect(ngFilter.ExcludeAll).To(Equal(withoutNodeGroup))
		
				Expect(cmd.ClusterConfig.NodeGroups).To(HaveLen(loaderTest.nodeGroupCount))
		
				_, err := cmd.NewCtl()
				Expect(err).ToNot(HaveOccurred())
		
			}
		}
```

Comment (errordeveloper) [reply pkg/ctl/cmdutils/configfile_test.go]: yeah, I thought about it, but didn't seem to make that much difference

**Pull request merged**